stdlib/Distributed/src/pmap.jl

```diff
@@ -256,13 +256,18 @@ julia> collect(c)
 """
 function head_and_tail(c, n)
     head = Vector{eltype(c)}(uninitialized, n)
-    s = start(c)
-    i = 0
-    while i < n && !done(c, s)
+    n == 0 && return (head, c)
+    i = 1
+    y = iterate(c)
+    y == nothing && return (resize!(head, 0), ())
+    head[i] = y[1]
+    while i < n
+        y = iterate(c, y[2])
+        y == nothing && return (resize!(head, i), ())
```

Comment:
are you making this intentionally type-unstable?

Reply:
Frankly I just wanted to get the tests to pass so I could start looking at the performance, though I don't think it matters too much since it's only used in pmap, which is already fairly complex.